packages/react-instantsearch-hooks-dom/src/ui/Highlight.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+import { cx } from './lib/cx';
+
+type HighlightPartProps = {
+  children: React.ReactNode;
+  baseClassName: string;
+  highlightedTagName: React.ReactType;
+  nonHighlightedTagName: React.ReactType;
+  isHighlighted: boolean;
+};
+
+function HighlightPart({
+  baseClassName,
+  children,
+  highlightedTagName,
+  isHighlighted,
+  nonHighlightedTagName,
+}: HighlightPartProps) {
+  const TagName = isHighlighted ? highlightedTagName : nonHighlightedTagName;
+
+  return (
+    <TagName
+      className={
+        isHighlighted
+          ? `${baseClassName}-highlighted`
+          : `${baseClassName}-nonHighlighted`
+      }
+    >
+      {children}
+    </TagName>
+  );
+}
+
+type HighlightedPart = {
+  isHighlighted: boolean;
+  value: string;
+};
+
+export type HighlightProps = React.HTMLAttributes<HTMLSpanElement> & {
+  baseClassName: string;
+  highlightedTagName?: React.ReactType;
+  nonHighlightedTagName?: React.ReactType;
+  separator?: React.ReactNode;
+  parts: HighlightedPart[];
+  partsMultiple: HighlightedPart[][];
+};
+
+export function Highlight({
+  baseClassName,
```

Comment:
A suit-like utility will probably be more useful because we can reuse it in other components, but it's not as greppable (but no solution erases the problem so that's not a dealbreaker IMO).

Are we currently happy with our suit utility in other flavors? Do we find it useful after using it for a while, or do we find it more of a needless indirection than a help?

Reply:
I've always found `suit` useful, but @dhayab mentioned he preferred as much strings as possible

Reply:
I also prefer regular strings, and the [grep test](http://jamie-wong.com/2013/07/12/grep-test/).

We could pass the full class names as props, instead of just the base.

Reply:
https://github.com/algolia/react-instantsearch/pull/3379